### PR TITLE
ci: Fix CI not testing with Postgres 16 and flaky test

### LINKED ISSUE #8613: Fix flaky Postgres tests
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

The flaky Postgres tests are currently the most significant bottleneck when it comes to merging PRs:

- More reviewer effort required to restart failing tests
- Longer wait time for contributors to see PRs getting merged
- Increased resource costs (GitHub Action compute time)

While there were other CI/CD related bottlenecks in the past, they have largely been removed over time so that this issue became now the most impactful one.

### Solution

Fix the flakiness.

### PR DESCRIPTION
**Pull request by dplewis**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Installed PG 15 instead of 16 as documented

Closes: https://github.com/parse-community/parse-server/issues/8613
Closes: https://github.com/parse-community/parse-server/pull/9107
Closes: https://github.com/parse-community/parse-server/pull/9110

## Approach
* Cleaned up TestUtils and Suite
* Attempt to remove flaky tests
* Add proper PG 16 image
* Update pg-promise to 11.9.1
* Update pg-monitor to 2.1.0

### CODE REVIEW

**Comment by dplewis** [comment]
@mtrezza I think this fixes that last flaky test

File: src/TestUtils.js [hunk 1]
```diff
@@ -26,3 +26,19 @@ export function destroyAllDataPermanently(fast) {
     })
   );
 }
+
+export function resolvingPromise() {
```
Comment (mtrezza) [inline src/TestUtils.js:30]: Interesting, do you know why moving it from helper to src fixed this?

Comment (dplewis) [reply src/TestUtils.js:30]: It doesn't I put it in the wrong place in my last PR. Forgot a TestUtils file already exists.

The fix is a combination of using a global spy for better control, removing unnecessary configuration and improving the flaky test which I documented.

Comment (mtrezza) [reply src/TestUtils.js:30]: Nice

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.3.0-alpha.3](https://github.com/parse-community/parse-server/releases/tag/7.3.0-alpha.3)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.3.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.3.0](https://github.com/parse-community/parse-server/releases/tag/7.3.0)

**Pull request merged**